### PR TITLE
Add freud recipe

### PR DESCRIPTION
**Pull request by bdice**
The freud library provides users the ability to analyze molecular dynamics and Monte Carlo simulation trajectories for advanced metrics such as the radial distribution function and various order parameters. I co-maintain the library with help from @vyasr and @joaander.
https://bitbucket.org/glotzer/freud/

We have previously released this library on our own anaconda channel, and would now like to add it to conda-forge to increase availability.

### CODE REVIEW

**Comment by bdice** [comment]
Currently, CircleCI fails because it used boost 1.66 instead of 1.67, which includes a necessary fix for GCC compilers: https://svn.boost.org/trac10/ticket/13386

This may be resolved by making the build use boost 1.67. However, we intend to remove boost as a dependency in an upcoming version of the package.

**Comment by bdice** [comment]
It doesn't seem like excluding boost v1.66 is possible, for some reason.

**Comment by joaander** [comment]
Apparently, conda-forge pinning forces the use of boost 1.66. 
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml

Is there any way to disable this pinning locally for a single package? Freud only uses the boost headers and does not link to the libraries.

Alternately, we could possibly work around this issue by installing and building with the conda-forge clangdev and libcxx instead of gcc. This workaround could be removed when conda-forge adopts a newer gcc compiler for the build toolchain.

**Comment by joaander** [comment]
@bdice you still have boost in the run dependency section. That is not needed as freud only uses the boost headers.

**Comment by joaander** [comment]
Weird, the new conda-forge-pinning 2018.06.19-1 is installed, but it still pulls in boost 1.66.0 during the conda-build process.

**Comment by bdice** [comment]
@conda-forge-admin, please rerender

**Comment by mbargull** [comment]
The bump to `boost 1.67` is not yet part of a released version/build of `conda-forge-pinning`. The current release is based on commit https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/db70a93213fd82e7deea878e9ca398d449dc47a6/recipe/conda_build_config.yaml#L228.

File: recipes/freud/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  license_family: BSD
+  license_file: LICENSE
+  summary: 'The freud library provides users the ability to analyze generic data from a variety of sources including simulation and experimental data for advanced metrics such as the radial distribution function and various order parameters.'
+  doc_url: http://{{ name }}.readthedocs.io/
+  dev_url: https://bitbucket.org/glotzer/{{ name }}
+
+extra:
+  recipe-maintainers:
+    - bdice
+    - vyasr
```
Comment (mariusvniekerk) [inline recipes/freud/meta.yaml]: @vyasr @joaander 
Are you okay with being a maintainer for this conda-forge recipe?  For details as to what this entails see [the docs](https://conda-forge.org/docs/recipe.html#maintainer-role)

Comment (joaander) [reply recipes/freud/meta.yaml]: Yes.

Comment (vyasr) [reply recipes/freud/meta.yaml]: Yes.

**Pull request merged**